codec-http/src/main/java/io/netty/handler/codec/http/HttpHeaderDateFormatter.java

```diff
@@ -0,0 +1,419 @@
+                if (currentPartValue == 0) {
+                    // invalid :: (nothing in between)
+                    return false;
+
+                } else if (currentPartNumber == 0) {
+                    // flushing hours
+                    localHours = currentPartValue;
+                    currentPartValue = 0;
+                    currentPartNumber++;
+
+                } else if (currentPartNumber == 1) {
+                    // flushing minutes
+                    localMinutes = currentPartValue;
+                    currentPartValue = 0;
+                    currentPartNumber++;
+
+                } else if (currentPartNumber == 2) {
+                    // invalid, too many :
+                    return false;
+                }
+            } else {
+                // invalid char
+                return false;
+            }
+        }
+
+        if (currentPartValue > 0) {
+            // pending seconds
+            localSeconds = currentPartValue;
+        }
+
+        if (localHours >= 0 && localMinutes >= 0 && localSeconds >= 0) {
+            hours = localHours;
+            minutes = localMinutes;
+            seconds = localSeconds;
+            return true;
+        }
+
+        return false;
+    }
+
+    private boolean tryParseDayOfMonth(CharSequence txt, int tokenStart, int tokenEnd) {
+        int len = tokenEnd - tokenStart;
+
+        if (len != 1 && len != 2) {
+            return false;
+        }
+
+        int localDayOfMonth = 0;
+        for (int i = tokenStart; i < tokenEnd; i++) {
+            char c = txt.charAt(i);
+            if (isDigit(c)) {
+                localDayOfMonth = localDayOfMonth * 10 + getNumericalValue(c);
+            } else {
+                // invalid
+                return false;
+            }
+        }
+
+        if (localDayOfMonth > 0) {
+            dayOfMonth = localDayOfMonth;
+            return true;
+        }
+
+        return false;
+    }
+
+    private boolean tryParseMonth(CharSequence txt, int tokenStart, int tokenEnd) {
+        int len = tokenEnd - tokenStart;
+
+        if (len == 3) {
+            String tokenString = txt.subSequence(tokenStart, tokenEnd).toString();
+
+            if (tokenString.equalsIgnoreCase("Jan")) {
+                month = Calendar.JANUARY;
+            } else if (tokenString.equalsIgnoreCase("Feb")) {
+                month = Calendar.FEBRUARY;
+            } else if (tokenString.equalsIgnoreCase("Mar")) {
+                month = Calendar.MARCH;
+            } else if (tokenString.equalsIgnoreCase("Apr")) {
+                month = Calendar.APRIL;
+            } else if (tokenString.equalsIgnoreCase("May")) {
+                month = Calendar.MAY;
+            } else if (tokenString.equalsIgnoreCase("Jun")) {
+                month = Calendar.JUNE;
+            } else if (tokenString.equalsIgnoreCase("Jul")) {
+                month = Calendar.JULY;
+            } else if (tokenString.equalsIgnoreCase("Aug")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "Aug" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1132)